utils/fileops.go

```diff
@@ -0,0 +1,81 @@
+package utils
+
+import (
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func StoreHTTPResponseToFile(r *http.Response, fileName string, dataFeedName string) error {
```

Comment:
It seems excessive to pass `dataFeedName` to this function if its only used in one error-case log message. Better to do the logging at the caller level that has the context directly.